client.js

```diff
@@ -326,34 +229,84 @@ async function downloadMod() {
+		if (entry.isSymbolicLink()) {
+			if (!await fs.pathExists(instance.path("mods", entry.name))) {
+				logger.log(`Removing broken symlink ${entry.name}`);
+				await fs.unlink(instance.path("mods", entry.name));
+			}
+		}
+	}
+
+	// Link entries that are in sharedMods but not in instance mods.
+	let instanceModsEntries = new Set(await fs.readdir(instance.path("mods")));
+	for (let entry of await fs.readdir(sharedMods, { withFileTypes: true })) {
+		if (entry.isFile()) {
+			if (['.zip', '.dat'].includes(path.extname(entry.name))) {
+				if (!instanceModsEntries.has(entry.name)) {
+					logger.log(`linking ${entry.name} from ${sharedMods}`);
+					let target = path.join(sharedMods, entry.name);
+					let link = instance.path("mods", entry.name);
+
+					if (process.platform !== "win32") {
+						await fs.symlink(path.relative(path.dirname(link), target), link);
+
+					// On Windows symlinks require elevated privileges, which is
+					// not something we want to have.  For this reason the mods
+					// are hard linked instead.  This has the drawback of not
+					// being able to identify when mods are removed from the
+					// sharedMods directory, or which mods are linked.
+					} else {
+						await fs.link(target, link);
+					}
+				}
+
+			} else {
+				logger.warning(`Warning: ignoring file '${entry.name}' in sharedMods`);
+			}
+
+		} else {
+			logger.warning(`Warning: ignoring non-file '${entry.name}' in sharedMods`);
+		}
+	}
+}
+
 async function createInstance(config, args, instance) {
-	// if instance does not exist, create it
-	console.log("Creating instance...");
-	fs.mkdirSync(instance.path());
-	fs.mkdirSync(instance.path("script-output"));
-	fs.writeFileSync(instance.path("script-output", "output.txt"), "");
-	fs.writeFileSync(instance.path("script-output", "orders.txt"), "");
-	fs.writeFileSync(instance.path("script-output", "txbuffer.txt"), "");
-	fs.mkdirSync(instance.path("mods"));
-	fs.mkdirSync(instance.path("instanceMods"));
-    fs.mkdirSync(instance.path("scenarios"));
-    ncp("lib/scenarios", instance.path("scenarios"), err => {
-        if (err) console.error(err)
-    });
-
-	// fs.symlinkSync('../../../sharedMods', instance.path("mods"), 'junction') // This is broken because it can only take a file as first argument, not a folder
-	fs.writeFileSync(instance.path("config.ini"), `[path]\r\n
-read-data=${ path.resolve(config.factorioDirectory, "data") }\r\n
-write-data=${ instance.path() }\r\n
-	`);
-
-	// this line is probably not needed anymore but Im not gonna remove it
-	fs.copySync('sharedMods', instance.path("mods"));
+	console.log(`Creating ${instance.path()}`);
+	await fs.ensureDir(instance.path());
+	await fs.ensureDir(instance.path("script-output"));
+
+	let target = path.join("lib", "scenarios");
+	let link = instance.path("scenarios");
+	await fs.symlink(path.relative(path.dirname(link), target), link, 'junction');
+
+	await symlinkMods(instance, "sharedMods", console);
 	let instconf = {
```

Comment:
Maybe use actual files with default values, so they are not defined in the code like this and loaded via require?